### DISCUSSION TITLE
book: queries for solving problems

### DISCUSSION THREAD

**Discussion by vanillajonathan** [PRQL/prql#3982] [Ideas]
There could be a page in the book that presents a list of problems along with PRQL queries to solve them.

This would aide in showing how PRQL is useful, and how you can apply PRQL to solve real-world problems!

Examples:
- A meteorologic institute wants to know which month has historically had the most rain.
- A carpool wants to know who utilizes the cars to drive the furthest distance.
- A climatology institute wants to know which year had the highest average, mean and median temperatures.
- A dating website wants to know the age of women that most men show interest in.
- The police wants to which cities of the last year had the highest occurrence of violent crime.
- A social network website wants to know which hashtags trended the last month.
- A online retailer wants to know which product category had the highest increase in sales during Christmas.
- A research institute wants to know which roads had the most night-time accidents.
- A power company wants to know which time of the day on average has the highest load. Weekdays only, weekends excluded.
- A call center wants to know which employee on average had the calls which lasted the shortest.
- A movie website wants to know which movies received the highest ratings from men over 40.
- A book website wants to know which genres of books are the most popular with females between the ages 13 and 19.
- A travel agency wants to know which destinations were the most popular in the month of June for the last three years.
- A modeling agency wants to know who were the most booked brunettes during this year's autumn months of September, October, and November.

Some of these questions might be silly or redundant, but the point is to show some good examples of questions and problems that might arise in the real world and show how they're solved using PRQL.

**Comment by hulxv** [comment]
It will be a good answer to the question, “Why would I use PRQL if I could use SQL that is supported by almost all databases?”
As mentioned in #3518, and after reading some opinions on [HR](https://news.ycombinator.com/item?id=37534773), most people don't know why PRQL was invented or it  is worth using
